File: frontend/src/types/matching.ts
export interface MatchResult {
  application_id: number | null
  role_id: number
  role_name: string | null
  posting_id: number
  company: string
  title: string | null
  deadline: string | null
  linkareer_url: string
  apply_url: string | null
  match_score: number
}

export interface ExplainResult {
  application_id: number
  explanation: string
}

File: backend/app/schemas/matching.py
from datetime import date

from pydantic import BaseModel


class MatchResult(BaseModel):
    application_id: int | None = None
    role_id: int
    role_name: str | None
    posting_id: int
    company: str
    title: str | None
    deadline: date | None
    linkareer_url: str
    apply_url: str | None
    match_score: int


class ExplainResponse(BaseModel):
    application_id: int
    explanation: str
